EventGenerator/src/MuCapGammaRayGenerator_tool.cc

```diff
@@ -0,0 +1,65 @@
+#include "art/Utilities/ToolMacros.h"
+
+#include "CLHEP/Random/RandFlat.h"
+
+#include "EventGenerator/inc/ParticleGeneratorTool.hh"
+
+#include "DataProducts/inc/PDGCode.hh"
+#include "MCDataProducts/inc/GenId.hh"
+#include "Mu2eUtilities/inc/RandomUnitSphere.hh"
+#include "Mu2eUtilities/inc/BinnedSpectrum.hh"
+#include "GlobalConstantsService/inc/GlobalConstantsHandle.hh"
+#include "GlobalConstantsService/inc/ParticleDataTable.hh"
+#include "GlobalConstantsService/inc/PhysicsParams.hh"
+
+namespace mu2e {
+  class MuCapGammaRayGenerator : public ParticleGeneratorTool {
+  public:
+    explicit MuCapGammaRayGenerator(Parameters const& conf) :
+      _pdgId(PDGCode::gamma),
+      _mass(GlobalConstantsHandle<ParticleDataTable>()->particle(_pdgId).ref().mass().value()),
+      _genId(GenId::MuCapGammaRayGenTool),
+      _energy(GlobalConstantsHandle<PhysicsParams>()->getCaptureGammaEnergy()),
+      _intensity(GlobalConstantsHandle<PhysicsParams>()->getCaptureGammaIntensity())
+    {
+
+    }
+      void generate(std::unique_ptr<GenParticleCollection>& out, const IO::StoppedParticleF& stop) override;
+
+    void setEngine(art::RandomNumberGenerator::base_engine_t& eng) {
+      _randomUnitSphere = new RandomUnitSphere(eng);
+      _randFlat = new CLHEP::RandFlat(eng);
+    }
+
+  private:
+    PDGCode::type _pdgId;
+    double _mass;
+    GenId _genId;
+    double _energy;
+    double _intensity;
+
+    SpectrumVar       _spectrumVariable;
+    BinnedSpectrum    _spectrum;
+
+    RandomUnitSphere*   _randomUnitSphere;
+    CLHEP::RandFlat* _randFlat;
+  };
+
+  void MuCapGammaRayGenerator::generate(std::unique_ptr<GenParticleCollection>& out, const IO::StoppedParticleF& stop) {
+    const CLHEP::Hep3Vector pos(stop.x, stop.y, stop.z);
+
+    double rand = _randFlat->fire();
+    if (rand < _intensity) {
```

Comment:
This code produces mean number of gammas per capture == intensity <= 1.     In our past simulations we assumed 2 gammas per capture.  I think the code has to change to sample the number of gammas from a Poisson distribution.

Reply:
This generator is for the 1.8 MeV gamma ray and is not given to the MuStopGenerator. It is here as an additional example that might be useful. The tool that we will use is MuCapPhotonGenerator and does the Poisson distribution with the assumed 2 photons per capture. I will rename this generator tool to MuCap1809keVGammaRayGenerator_tool or something similar to avoid confusion.